### PR TITLE
DOCS-#3098: Adding Intel distribution of Modin section

### LINKED ISSUE #3098: Add AI Kit Intel Distribution of Modin Section to Modin Documentation
@aregm 

Create separate section on open-source Modin documentation highlighting AI Kit Intel Distribution of Modin optimizations, AI Kit, and OmniSci backend with deeper explanation on how this differs from other Modin installs (beyond just install directions)

### PR DESCRIPTION
**Pull request by Garra1980**
Signed-off-by: izamyati <igor.zamyatin@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3098 
- [ ] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by Garra1980** [comment]
cc @raoberman

File: docs/installation.rst [hunk 1]
```diff
@@ -91,6 +94,13 @@ or explicitly
 
   conda install -c conda-forge modin-ray modin-dask modin-omnisci
 
+Using Intel\ |reg| Distribution of Modin
+"""""""""""""""""""""""""""""""""""""""
+
+With ``conda`` it is possible to install special version of Modin, part of Intel\ |reg| oneAPI AI Analytics Toolkit.
+This version of Modin powered by :doc:`OmniSci</UsingOmnisci/index>` backend that contains a bunch of
```
Comment (YarShev) [inline docs/installation.rst:101]:
```suggestion
This version of Modin powered by :doc:`OmniSci</UsingOmnisci/index>` engine that contains a bunch of
```

File: docs/installation.rst [hunk 2]
```diff
@@ -91,6 +94,13 @@ or explicitly
...
+optimizations for Intel hardware. All details can be found here_.
```
Comment (YarShev) [inline docs/installation.rst:102]:
```suggestion
optimizations for Intel hardware. All details can be found on `Intel Distribution of Modin`_ page.
```

File: docs/installation.rst [hunk 3]
```diff
@@ -151,3 +161,5 @@ Once cloned, ``cd`` into the ``modin`` directory and use ``pip`` to install:
 .. _Ray: http://ray.readthedocs.io
 .. _Dask: https://github.com/dask/dask
 .. _OmniSci: https://www.omnisci.com/platform/omniscidb
+.. _here: https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/distribution-of-modin.html#gs.86stqv
```
Comment (YarShev) [inline docs/installation.rst:164]:
```suggestion
.. _`Intel Distribution of Modin`: https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/distribution-of-modin.html#gs.86stqv
```

File: docs/installation.rst [hunk 4]
```diff
@@ -91,6 +94,13 @@ or explicitly
 
   conda install -c conda-forge modin-ray modin-dask modin-omnisci
 
+Using Intel\ |reg| Distribution of Modin
+"""""""""""""""""""""""""""""""""""""""
+
+With ``conda`` it is possible to install special version of Modin, part of Intel\ |reg| oneAPI AI Analytics Toolkit.
+This version of Modin powered by :doc:`OmniSci</UsingOmnisci/index>` engine that contains a bunch of
```
Comment (vnlitvinov) [inline docs/installation.rst:101]:
```suggestion
This version of Modin is powered by :doc:`OmniSci</UsingOmnisci/index>` engine that contains a bunch of
```

**Comment by raoberman** [comment]
@Garra1980  Phrasing edit: With conda it is also possible to install Intel® Distribution of Modin, a special version of Modin that is part of[ Intel® oneAPI AI Analytics Toolkit.](https://software.intel.com/content/www/us/en/develop/tools/oneapi/ai-analytics-toolkit.html#gs.88nh8y) This version of Modin is powered by :doc:`OmniSci</UsingOmnisci/index>` engine that contains a bunch of optimizations for Intel hardware. More details can be found on Intel® Distribution of Modin page.

**Review by vnlitvinov — DISMISSED** [review]
LGTM

**Comment by Garra1980** [comment]
> @Garra1980 Phrasing edit: With conda it is also possible to install Intel® Distribution of Modin, a special version of Modin that is part of[ Intel® oneAPI AI Analytics Toolkit.](https://software.intel.com/content/www/us/en/develop/tools/oneapi/ai-analytics-toolkit.html#gs.88nh8y) This version of Modin is powered by :doc:`OmniSci</UsingOmnisci/index>` engine that contains a bunch of optimizations for Intel hardware. More details can be found on Intel® Distribution of Modin page.

Done

**Review by YarShev — APPROVED** [review]

**Pull request merged**